log/tm_logger.go

```diff
@@ -0,0 +1,77 @@
+package log
+
+import (
+	"fmt"
+	"io"
+
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/go-kit/kit/log/term"
+)
+
+const (
+	msgKey = "msg"
+)
+
+type tmLogger struct {
+	srcLogger kitlog.Logger
+}
+
+// NewTmTermLogger returns a logger that encodes msg and keyvals to the Writer
+// using go-kit's log as an underlying logger and our custom formatter. Note
+// that underlying logger could be swapped with something else.
+func NewTmLogger(w io.Writer) Logger {
+	// Color by level value
+	colorFn := func(keyvals ...interface{}) term.FgBgColor {
+		for i := 0; i < len(keyvals)-1; i += 2 {
```

Comment:
yeah, that's probably not good for performance. I might know the way how to fix that (have index keyval always first)